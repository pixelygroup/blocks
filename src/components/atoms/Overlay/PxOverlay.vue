<template>
  <transition name="fade">
    <div v-show="showOverlay" class="px-overlay" @click.stop="handleClick" />
  </transition>
</template>

<script>
export default {
  name: 'PxOverlay',

  props: {
    showOverlay: Boolean,
  },

  methods: {
    handleClick() {
      this.$emit('close')
    },
  },
}
</script>

<style lang="scss" scoped>
@import '../../../sass/framework/framework.scss';

.px-overlay {
  background-color: rgba(44, 62, 73, 0.55);
  bottom: 0;
  display: block;
  height: 100%;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  width: 100%;
  z-index: $zindex-modal-overlay;
}

.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.5s;
}

.fade-enter,
.fade-leave-to {
  opacity: 0;
}
</style>
