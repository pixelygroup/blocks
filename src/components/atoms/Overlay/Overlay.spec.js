import { mount, shallowMount } from '@vue/test-utils'
import PxOverlay from './PxOverlay.vue'

describe('Overlay', () => {
  it('is a Vue instance', () => {
    const wrapper = mount(PxOverlay)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders correctly', () => {
    const wrapper = mount(PxOverlay)
    expect(wrapper).toMatchSnapshot()
  })

  it('is initially hidden', () => {
    const wrapper = shallowMount(PxOverlay)
    expect(wrapper.props('showOverlay')).toBe(false)
    expect(wrapper.isVisible()).toBe(false)
  })

  it('should display the overlay on showOverlay=true', () => {
    const wrapper = mount(PxOverlay)
    wrapper.setProps({ showOverlay: true })
    expect(wrapper.isVisible()).toBe(true)
  })

  it('should emit close event', () => {
    const wrapper = mount(PxOverlay)
    wrapper.setProps({ showOverlay: true })
    wrapper.trigger('click')
    expect(wrapper.emitted('close')).toBeTruthy()
  })
})
