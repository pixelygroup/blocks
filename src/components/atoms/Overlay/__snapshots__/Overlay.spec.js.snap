// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay renders correctly 1`] = `<div class="px-overlay" style="display: none;" name="fade"></div>`;
